utilities/user_utils.py

```diff
@@ -28,111 +25,24 @@ class UserTestSession:
     secret_name: str
     username: str
     password: str
-    user_context: str
-    original_context: str
+    original_user: str
+    api_server_url: str
 
     def __post_init__(self) -> None:
         """Validate the session data after initialization."""
         if not all([self.idp_name, self.secret_name, self.username, self.password]):
             raise ValueError("All session fields must be non-empty")
-        if not self.user_context or not self.original_context:
-            raise ValueError("Both user and original contexts must be set")
+        if not (self.api_server_url and self.original_user):
+            raise ValueError("Original user information and api url must be set")
 
     def cleanup(self) -> None:
         """Clean up the user context."""
-        if self.user_context:
-            # First switch to original context
-            run_command(command=["oc", "config", "use-context", self.original_context], check=True)
-            LOGGER.info(f"Switched to original context: {self.original_context}")
-            # Then delete the user context
-            LOGGER.info(f"Deleting user context: {self.user_context}")
-            run_command(
-                command=["oc", "config", "delete-context", self.user_context],
-                check=False,  # Don't fail if context doesn't exist
-            )
-
-
-@contextmanager
-def create_test_idp(
-    idp_name: str = "test-htpasswd-idp", secret_name: str = "test-htpasswd-secret"
-) -> Generator[UserTestSession, None, None]:
-    """
-    Context manager to create and manage a test HTPasswd IDP in OpenShift.
-    Creates the IDP and test user, then cleans up after use.
-
-    Args:
-        idp_name: Name for the IDP
-        secret_name: Name for the secret
+        user = User(name=self.username)
+        if user.exists:
+            user.delete()
```

Comment:
shouldn't this be handled by the teardown of the fixture creating the user in the first place?

Reply:
I don't see it happening, I saw a bunch of leftover users instead.